bin/athena.bb

```
@@ -0,0 +1,106 @@
+#!/usr/bin/env bb
+
+(ns athena
+  (:require
+    [babashka.cli :as cli]
+    [babashka.curl :as curl]
+    [babashka.process :as p]
+    [clojure.data.xml :as x]
+    [clojure.java.io :as io]
+    [clojure.string :as str]
+    [clojure.zip :as z]))
+
+;; Since amazon is still not putting their athena jdbc on to maven, the easiest thing for us to do is setup a maven-y s3 repo and copy their jar into it.
+;;
+;; This script allows you to run:
+;; ./bin/athena.bb <version> [--deploy]
+;;
+;; For example:
+;; ./bin/athena.bb 2.0.35 --deploy
+;;
+;; Will download the 2.0.35 jar from amazon's release bucket, create and update the necessary maven repo files, then upload all those files to our maven repo.
+;;
+;; It uses the jar build WITHOUT the AWS SDK.
+;;
+;; You must have bb (babashka), aws (awscli), and shasum commands available. You must have aws credentials available, e.g. through aws configure
+
+(defn find-pred [z p?]
+  (loop [z z]
+    (cond
+      (or (nil? z) (z/end? z))
+      nil
+
+      (-> z z/node p?)
+      z
+
+      :else
+      (recur (z/next z)))))
+
+(defn topmost [z]
+  (loop [z z]
+    (if-let [parent (z/up z)]
+      (recur parent)
+      z)))
```

In this program, isn't this the same as
```suggestion
(def topmost (comp z/xml-zip z/root)
```